system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -195,14 +195,16 @@ class XpathContextState
 
 typedef std::vector<XpathContextState> XPathContextStateVector;
 
-class CLibXpathContext : public CInterfaceOf<IXpathContext>
+class CLibXpathContext : public CInterfaceOf<IXpathContext>, implements IVariableSubstitutionHelper
 {
 public:
     XPathInputMap provided;
     xmlDocPtr m_xmlDoc = nullptr;
     xmlXPathContextPtr m_xpathContext = nullptr;
     ReadWriteLock m_rwlock;
```

Comment:
Which operations does this need to protect? Only some of the methods in this class use it. Some do a fair amount of work before calling `setLocation()` and others like `rename()` and `remove()` don't appear to use it at all, which seems like it might be an oversight.

Reply:
I may just remove this in the future.  The scripting is not multithreaded and would have to be majorly refactored to make it so.  It's not a declarative language.